toxcore/net_crypto.c

```diff
@@ -1863,30 +1818,14 @@ static int create_crypto_connection(Net_Crypto *c)
     }
 
     if (id != -1) {
-        pthread_mutex_t *mutex = (pthread_mutex_t *)mem_alloc(c->mem, sizeof(pthread_mutex_t));
-
-        if (mutex == nullptr) {
-            pthread_mutex_unlock(&c->connections_mutex);
-            return -1;
-        }
-
-        if (pthread_mutex_init(mutex, nullptr) != 0) {
-            mem_delete(c->mem, mutex);
-            pthread_mutex_unlock(&c->connections_mutex);
-            return -1;
-        }
-
         // Memsetting float/double to 0 is non-portable, so we explicitly set them to 0
         c->crypto_connections[id].packet_recv_rate = 0.0;
         c->crypto_connections[id].packet_send_rate = 0.0;
         c->crypto_connections[id].last_packets_left_rem = 0.0;
         c->crypto_connections[id].packet_send_rate_requested = 0.0;
         c->crypto_connections[id].last_packets_left_requested_rem = 0.0;
-        c->crypto_connections[id].mutex = mutex;
-        c->crypto_connections[id].status = CRYPTO_CONN_NO_CONNECTION;
```

Comment:
Is this intentional? This is the only place where `crypto_connection` is set to this state.

Reply:
`#define CRYPTO_CONN_NO_CONNECTION 0`
It used to be defined as the zero.

Reply:
The mutex is very intentional, status I will check.

Reply:
let me check. this could be a mistake. since this line is there in my fork and in toktok master ...

Reply:
ok i see what happend here:
first the commit 6338cb46ad99dbc2806976aefabbf2e98476ebda
refactored the function to put c->crypto_connections[id].status = CRYPTO_CONN_NO_CONNECTION; in a different place.
then after that commit 4db6cfb19239a0d8b83ef91b1cc26fab1979fca7 most likely removed the line by mistake in a merge conflict resolve

Reply:
After looking at the code, I think we dont need that enum value.
`CRYPTO_CONN_FREE` represents the same thing. But now we need to make sure, we don't improperly reuse the connection without memsetting it...

Reply:
lets put this line back please

Reply:
ah nice, github did not update @zoff99 in my open page. Yes, lets add the line back in and figure out if we need `CONN_NO_CONNECTION` at all somewhere else.